YogaGlo/serverless-image-resizing
# Updating cloudformation template

The original plan was to have the cloudformation templates live the `yg-devops` repo alongside the other cloudformation templates. However this template needs to live next to the actual lambda function so that the compiled lambda package can be properly uploaded. So... I think it makes sense that the cloudformation template lives next to the code in this repo.

## Changes:
- updated cloudformation template to support the resources required by the modified lambda script in (https://github.com/YogaGlo/serverless-image-resizing/pull/1)
- updated the deploy script to pass the required parameters to the cloudformation template

## Notes: 
- this hasn't been fully tested as I do not have the proper permissions to create IAM roles
- I'm not sure if we can use the existing S3 buckets `ygthumbseq` and `ygthumbxxd` as they were not created by cloudformation. We may need to:
    1. create new s3 buckets and copy the contents from the old buckets
    1. manually create the lambda function to use the old buckets, however that defeats the purpose of the cloudformation template

## Testing Steps:
- compile the lambda package by running `make dist`
- ensure you have the [awscli](https://aws.amazon.com/cli/) installed. `aws --version` (version at least 1.11.19)
- ensure your aws credentials are in your `ENV` vars, or set properly using `aws configure` on the command line
- run `bin/deploy`
- provide answers when prompted
- marvel in your miraculous success and/or horrid failure
